dotCMS/src/integration-test/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImplTest.java

```diff
@@ -1787,4 +1792,173 @@ public void findDeletedContentletByIdentifierAnyLanguage()
 
         assertNull(contentletByIdentifierAnyLanguage);
     }
+
+    /**
+     * Method to test: {@link ESContentletAPIImpl#checkin(Contentlet, User, boolean)}
+     * When:
+     * - Create two {@link Language}: A and B
+     * - Create two {@link Template}: A and B
+     * - Create a {@link HTMLPageAsset} in the first language.
+     * - Create a new version of the same {@link HTMLPageAsset} but with different {@link Language}
```

Comment:
It's missing saying info about the page and template

Reply:
done https://github.com/dotCMS/core/pull/23577/commits/c88aeb826bfdbcbcf5fb893b31972ea4392000f7#diff-093de07a53a725b587963fd60c024f52e089c4f78e7ea88ba8e3a3a0cdaecab9R1801